app/lib/handlers.dart

```diff
@@ -448,6 +453,20 @@ Future<shelf.Response> apiPackagesHandler(shelf.Request request) async {
   return _jsonResponse(json);
 }
 
+/// Handles requests for /flutter
+Future<shelf.Response> flutterHandler(shelf.Request request) async {
+  final String path = request.requestedUri.path;
+  final String basePath = '/flutter/plugins';
+  if (path == '/flutter' || path == '/flutter/') {
```

Comment:
If this function gets called for `/flutter/foo/bar?page=2` we will treat it as `/flutter/plugins?page=2`. We should rather return a 404 in such a case.

The functionality of this function should be, if I see this correctly, only handling URLs of the form:
```
/flutter/plugins
/flutter/plugins?page=<number>
```

See comment above.